cpp/src/arrow/compute/kernels/scalar_cast_test.cc

```diff
@@ -197,8 +197,8 @@ TEST(Cast, CanCast) {
 
   ExpectCanCast(utf8(), {timestamp(TimeUnit::MILLI)});
   ExpectCanCast(large_utf8(), {timestamp(TimeUnit::NANO)});
-  ExpectCannotCast(timestamp(TimeUnit::MICRO),
-                   kBaseBinaryTypes);  // no formatting supported
+  // ExpectCannotCast(timestamp(TimeUnit::MICRO),
+  //                 kBaseBinaryTypes);  // no formatting supported
```

Comment:
This tests was commented because it fails with the new changes. According to the fail log the conversion from timestamp[us] to string is expected to be false. I'm not sure how to proceed here.

Reply:
Because we have this above:

```cpp
  for (const std::shared_ptr<DataType>& in_ty : TemporalTypes()) {
```

we're actually adding kernels not just for date32/date64, but also for timestamp. So I think this can be changed to `ExpectCanCast`. And then we can add test cases below for timestamp as well.

Reply:
Fixed, thanks.